src/persistenttabmodel.cpp

```diff
@@ -0,0 +1,150 @@
+}
+
+PersistentTabModel::~PersistentTabModel()
+{
+}
+
+void PersistentTabModel::tabsAvailable(QList<Tab> tabs)
+{
+    beginResetModel();
+    int oldCount = count();
+    m_tabs.clear();
+    m_tabs = tabs;
+
+    if (m_tabs.count() > 0) {
+        QString activeTabId = DBManager::instance()->getSetting("activeTabId");
+        bool ok = false;
+        int tabId = activeTabId.toInt(&ok);
+        int index = findTabIndex(tabId);
+        if (index >= 0) {
+            m_activeTab = m_tabs[index];
+        } else {
+            // Fallback for browser update as this "activeTabId" is a new setting.
+            m_activeTab = m_tabs[0];
+        }
+        emit activeTabIndexChanged();
+    } else {
+        emit tabsCleared();
+    }
+
+    endResetModel();
+
+    if (count() != oldCount) {
+        emit countChanged();
+    }
+
+    int maxTabId = DBManager::instance()->getMaxTabId();
+    if (m_nextTabId != maxTabId + 1) {
+        m_nextTabId = maxTabId + 1;
+        emit nextTabIdChanged();
+    }
+
+    // Startup should be synced to this.
+    if (!m_loaded) {
+        m_loaded = true;
+        emit loadedChanged();
+    }
+}
+
+void PersistentTabModel::tabChanged(const Tab &tab)
+{
+#if DEBUG_LOGS
+    qDebug() << "new tab data:" << &tab;
+#endif
+    if (m_tabs.isEmpty()) {
+        qWarning() << "No tabs!";
+        return;
+    }
+
+    int i = findTabIndex(tab.tabId());
+    if (i > -1) {
+        QVector<int> roles;
+        Tab oldTab = m_tabs[i];
+        if (oldTab.url() != tab.url()) {
+            roles << UrlRole;
+        }
+        if (oldTab.title() != tab.title()) {
+            roles << TitleRole;
+        }
+        if (oldTab.thumbnailPath() != tab.thumbnailPath()) {
+            roles << ThumbPathRole;
+        }
+        m_tabs[i] = tab;
+        QModelIndex start = index(i, 0);
+        QModelIndex end = index(i, 0);
+        emit dataChanged(start, end, roles);
+    }
+
+    if (tab.tabId() == m_activeTab.tabId()) {
+        m_activeTab = tab;
+        emit activeTabChanged(tab.tabId(), tab.tabId(), true);
+    }
+}
+
+int PersistentTabModel::createTab() {
+    return DBManager::instance()->createTab();
+}
+
+int PersistentTabModel::createLink(int tabId, QString url, QString title) {
+    return DBManager::instance()->createLink(tabId, url, title);
+}
+
+void PersistentTabModel::updateTitle(int tabId, int linkId, QString url, QString title)
+{
+    DBManager::instance()->updateTitle(tabId, linkId, url, title);
+}
+
+void PersistentTabModel::removeTab(int tabId)
+{
+    DBManager::instance()->removeTab(tabId);
+}
+
+int PersistentTabModel::nextLinkId() {
+    return DBManager::instance()->nextLinkId();
+}
+
+void PersistentTabModel::updateTab(int tabId, QString url, QString title, QString path) {
+    DBManager::instance()->updateTab(tabId, url, "", "");
+}
+
+void PersistentTabModel::navigateTo(int tabId, QString url, QString title, QString path) {
+    DBManager::instance()->navigateTo(tabId, url, "", "");
```

Comment:
I've noticed build warnings about unused title and path. Could you please add Q_UNUSED() to updateTab() and navigateTo()?